globals.d.ts

```diff
@@ -0,0 +1,29 @@
+declare interface IContract {
+  address: string;
+  new: (...params: any[]) => Promise<IContractInstance>;
+}
+
+declare interface IContractInstance {
+  address: string;
+}
+
+declare interface ISignHash extends IContractInstance {
+  sign: (hash: string) => Promise<void>;
+  getSigners: (hash: string) => Promise<string[]>;
+}
+
+declare interface IArtifacts {
+  require: (path: string) => IContract;
```

Comment:
I think we can take more advantage of TS here using generic types and literal typing. Please checkout 
https://github.com/biern/truffle-typescript-example/blob/master/contracts/truffle.d.ts

Reply:
Thanks! I'm still learning TS, I made changes inspired by your example.